netket/models/autoreg.py

```diff
@@ -44,8 +45,8 @@ def conditionals(self, inputs: Array, cache: PyTree) -> Tuple[Array, PyTree]:
 class ARNNDense(ARNN):
     """Autoregressive neural network with dense layers."""
 
-    hilbert_local_size: int
-    """local size of the Hilbert space."""
+    hilbert: CustomHilbert
+    """the discrete Hilbert space."""
```

Comment:
Does this actually work?
I think this should be a `pytree_node=False` . I'm surprised it works without...

Reply:
At first I only wanted to pass `hilbert.local_size` as an int, so I could know the number of channels that the network should output, but Giuseppe suggested to pass the whole `hilbert`.

Do you think it's necessary to pass the whole `hilbert` into the model, and is there a safe way to do that?

Reply:
I am in favour of passing the whole hilbert.
It's just that done like this it might play badly with jax...

Can you try declaring this field as 

```suggestion
    hilbert: CustomHilbert = struct.field(pytree_node=False)
    """the discrete Hilbert space."""
```

Reply:
ah, no! wait. All is fine.
I had already implemented stuff to do this.
Ignore my suggestion